src/main/java/longah/LongAh.java

```diff
@@ -1,21 +1,173 @@
+     *
+     * @param person1 The name of the first person.
+     * @param amount  The amount of the debt.
+     * @param person2 The name of the second person.
+     */
+    public void addDebt(String person1, double amount, String person2) {
+        Member from = getOrCreateMember(person1);
+        Member to = getOrCreateMember(person2);
+        transactions.add(new Transaction(from, to, amount));
+    }
+
+    /**
+     * Lists all debts between members.
+     */
+    public void listAllDebts() {
+        Map<String, Double> balances = transactions.calculateBalances();
+
+        // Display simplified debts
+        boolean hasDebts = false;
+        for (String name : balances.keySet()) {
+            double amount = balances.get(name);
+            if (amount != 0) {
+                String formattedAmount = String.format("%.2f", Math.abs(amount));
+                String otherPerson = getOtherPerson(name);
+                if (amount > 0 && balances.containsKey(otherPerson) && balances.get(otherPerson) == -amount) {
+                    System.out.println(otherPerson + " owes " + name + " $" + formattedAmount);
+                    hasDebts = true;
+                } else if (amount < 0 && balances.containsKey(otherPerson) && balances.get(otherPerson) == -amount) {
+                    // Skip, as the other person already owes this person
+                } else {
+                    System.out.println(name + " owes " + otherPerson + " $" + formattedAmount);
+                    hasDebts = true;
+                }
+            }
+        }
+
+        if (!hasDebts) {
+            System.out.println("No debts, Huat Ah!");
+        }
+    }
+
+    /**
+     * Gets the other person involved in a transaction.
+     *
+     * @param name The name of the person.
+     * @return The name of the other person.
+     */
+    private String getOtherPerson(String name) {
+        return transactions.getOtherPerson(name);
+    }
+
+    /**
+     * Deletes a debt transaction by index.
+     *
+     * @param index The index of the transaction to delete.
+     */
+    public void deleteDebt(int index) {
+        transactions.remove(index);
+    }
+
+    /**
+     * Finds debts involving a specific person.
+     *
+     * @param person The name of the person to find debts for.
+     */
+    public void findDebts(String person) {
+        for (Transaction transaction : transactions.getTransactions()) {
+            if (transaction.getInvolves(person)) {
+                System.out.println(transaction);
+            }
+        }
+    }
+
+    /**
+     * Clears all debts.
+     */
+    public void clearAllDebts() {
+        transactions.clear();
+    }
+
+    /**
+     * Gets or creates a member with the given name.
+     *
+     * @param name The name of the member.
+     * @return The Member object.
+     */
+    private Member getOrCreateMember(String name) {
+        return members.computeIfAbsent(name, Member::new);
+    }
+
```

Comment:
All debt and member methods do not belong here. Should go in either `GroupList` or `TransactionList`.